File: CPP/USAddressBuilder.cpp
//
// Created by Nutzer on 22.08.2025.
//

#include "USAddressBuilder.h"

void USAddressBuilder::buildLine3() {
    address.addLine(addressData.city + ", " + addressData.region  + " "  + addressData.zip + "\n");
}

USAddressBuilder::USAddressBuilder(const address::AddressData& address_data)
    :AddressBuilder(address_data){
    title_f = "Ms";
    title_m = "Mr";
}

File: Python/USAddressBuilder.py
from AddressBuilder import AddressBuilder
from AddressData import AddressData


class USAddressBuilder(AddressBuilder):
    def __init__(self, address_data: AddressData):
        super().__init__(address_data)
        self.title_f = "Ms"
        self.title_m = "Mr"

    def build_line3(self) -> None:
        self.address.add_line(self.address_Data.city + ", " + self.address_Data.region + " " + self.address_Data.zip + "\n")
